src/Writer/Extension/PodcastIndex/Feed.php

```diff
@@ -106,6 +110,189 @@ public function setPodcastIndexFunding(array $value): Feed
+            );
+        }
+        if (! is_string($value['identifier'])) {
+            throw new Writer\Exception\InvalidArgumentException(
+                'invalid parameter: "identifier" of "license" must be of type string.'
+            );
+        }
+        if (! is_string($value['url']) || ! filter_var($value['url'], FILTER_VALIDATE_URL)) {
+            throw new Writer\Exception\InvalidArgumentException(
+                'invalid parameter: "url" of "license": must be a url starting with "http://" or "https://"'
+            );
+        }
+        $this->data['license'] = $value;
+        return $this;
+    }
+
+    /**
+     * Set feed location
+     *
+     * @param array $value [description: string, geo: string|null, osm: string|null]
+     * @return $this
+     * @throws Writer\Exception\InvalidArgumentException
+     */
+    public function setPodcastIndexLocation(array $value)
+    {
+        if (! isset($value['description'])) {
+            throw new Writer\Exception\InvalidArgumentException(
+                'invalid parameter: "location" must be an array containing at least the key "description" (node value)'
+            );
+        }
+        if (! is_string($value['description'])) {
+            throw new Writer\Exception\InvalidArgumentException(
+                'invalid parameter: key "description" of "location" must be of type string.'
+            );
+        }
+        if (isset($value['geo']) && ! is_string($value['geo'])) {
+            throw new Writer\Exception\InvalidArgumentException(
+                'invalid parameter: key "geo" of "location" must be of type string. example: "geo:-27.86159,153.3169"'
+            );
+        }
+        if (isset($value['osm']) && ! is_string($value['osm'])) {
+            throw new Writer\Exception\InvalidArgumentException(
+                'invalid parameter: key "osm" of "location" must be of type string. example: "W43678282"'
+            );
+        }
+        $this->data['location'] = $value;
+        return $this;
+    }
+
+    /**
+     * Set feed images
+     *
+     * @param array $value [scrset: string]
+     * @return $this
+     * @throws Writer\Exception\InvalidArgumentException
+     */
+    public function setPodcastIndexImages(array $value)
+    {
+        if (! isset($value['srcset'])) {
+            throw new Writer\Exception\InvalidArgumentException(
+                'invalid parameter: "images" must be an array containing the key "srcset"'
+            );
+        }
+        if (! is_string($value['srcset'])) {
+            throw new Writer\Exception\InvalidArgumentException(
+                'invalid parameter: key "srcset" of "images" must be of type string containing comma-seperated urls'
+            );
+        }
+        $this->data['images'] = $value;
+        return $this;
+    }
+
+    /**
+     * Set feed update frequency
+     *
+     * @param array $value [description: string, complete: bool|null, dtstart: string|null, rrule: string|null]
```

Comment:
Optional keys in array shapes are `array{optional?: type}` if it's OK for the key to contain null, it'd be `array{optional?: type|null}`

```suggestion
     * @param array{description: string, complete?: bool, dtstart?: string, rrule?: string} $value
```